.github/workflows/CI.yaml

```
@@ -10,65 +10,90 @@ env:
   TEAMCITY_DEV_URL: ${{ secrets.TEAMCITY_DEV_URL }}
   TEAMCITY_USER: ${{ secrets.TEAMCITY_USER }}
   TEAMCITY_PASSWORD: ${{ secrets.TEAMCITY_PASSWORD }}
-  SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
 
 jobs:
-  snyk-test:
+  compile:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: snyk/actions/setup@master
-      - name: Set up JDK 17
-        uses: actions/setup-java@v2
-        with:
-          java-version: '17'
-          distribution: 'adopt'
-      - name: Snyk test dependencies
-        run: snyk test --all-projects --severity-threshold=medium --fail-on=all
-
-  snyk-monitor:
-    runs-on: ubuntu-latest
-    if: github.event_name != 'pull_request'
-    steps:
-      - uses: actions/checkout@v2
-      - uses: snyk/actions/setup@master
-      - name: Set up JDK 17
-        uses: actions/setup-java@v2
-        with:
-          java-version: '17'
-          distribution: 'adopt'
-      - name: Snyk monitor dependencies
-        run: snyk monitor --all-projects --target-reference=${GITHUB_REF}
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-jdk
+      - uses: ./.github/actions/setup-gradle-cache
+      - name: Compile
+        run: ./gradlew compileJava compileTestJava
 
-  build:
+  code-ql:
     runs-on: ubuntu-latest
+    needs: compile
 
     # required by CodeQL
     permissions:
       security-events: write
       actions: read
 
-    env:
-      DOCKER_ENTERPRISE_DEV_URL: ${{ secrets.DOCKER_ENTERPRISE_DEV_URL }}
-      DOCKER_COMMUNITY_DEV_URL: ${{ secrets.DOCKER_COMMUNITY_DEV_URL }}
-      ENTERPRISE_TAR: enterprise-docker.tar
-      COMMUNITY_TAR: community-docker.tar
-
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-jdk
+      - uses: ./.github/actions/setup-gradle-cache
       - name: Initialize CodeQL
         uses: github/codeql-action/init@v2
         with:
           languages: java
+      - name: Compile
+        run: ./gradlew compileJava compileTestJava
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v2
+
+  common-tests:
+    runs-on: ubuntu-latest
+    needs: compile
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run common tests
+        uses: ./.github/actions/test-gradle-project
+        with:
+          project-name: common
```

```suggestion
    env:
      PROJECT_NAME: "core"
    steps:
      - uses: actions/checkout@v3
      - name: Run ${{env.PROJECT_NAME}} tests
        uses: ./.github/actions/test-gradle-project
        with:
          project-name: $PROJECT_NAME
```
Personally I would write it like this, i.e. the project name as a variable,
so as to avoid possible discrepancies between `name:`, `project-name`, and any lines that may be inserted in the future.

 But it's subjective, it's up to you if you want to change it or not.

---

Good point. I've simplified the repeated code with a matrix.